Homework20: Vladislav Savchuk

VladislavSavchuk opened the pull request:
# Developed program tests from homework #11

- Bank deposit
- Library

1. Created a separate module for tests.
2. Imported functions from the module.
3. Wrote tests to check the application performance.
4. The tests check both positive and negative results.
5. The tests are run using _unittest_

VladislavSavchuk committed:
add task 1

VladislavSavchuk committed:
add logs in .gitignore and loguru in requirements

VladislavSavchuk committed:
add init files in directories

VladislavSavchuk committed:
add logging in tests + little fix of bank_deposit

VladislavSavchuk committed:
add solution task #2 + little fix of library

VladislavSavchuk committed:
update command ruff <path>

Stml89 commented on homework/homework11/library.py:
```diff
@@ -20,18 +20,18 @@ def __str__(self):
             self.reserved = True
             self.reserved_by = user
-            return "Book has been successfully reserved"
+            return "reserved_success"
```
```suggestion
            return True
```

Stml89 commented on homework/homework11/library.py:
```diff
@@ -41,39 +41,36 @@ def receiving(self, user):
             self.reserved = False
             self.reserved_by = None
-            return "The book is free. You can take the book"
+            return "taken_success"
```
```suggestion
            return True
```

Stml89 commented on homework/homework11/bank_deposit.py:
```diff
@@ -31,12 +31,12 @@ def __init__(self, name: str):
+# print(person_2.name)
+# deposit_person2 = Bank(999999.99, 60, 10)
+# print(deposit_person2)
+# print(deposit_person2.calculate_monthly_capitalization())
```
If you do not want to use something when module was imported - just use `if __name__ == __main__`. And start using logging module instead of `print`

Stml89 commented on homework/homework20/test_bank_deposit.py:
```diff
@@ -0,0 +1,120 @@
+        bank = Bank(1500, 24, 3.5)
+        self.assertEqual(bank.deposit_amount, 1500)
+        self.assertEqual(bank.period, 24)
+        self.assertEqual(bank.rate, 3.5)
```
add some log that says that test was successful, e.g.:
`logger.info("Successfully verified  <.......>")`

Stml89 commented on homework/homework20/test_bank_deposit.py:
```diff
@@ -0,0 +1,120 @@
+        # Test with zero rate
+        bank = Bank(1000, 12, 0)
+        result = bank.calculate_monthly_capitalization()
+        self.assertEqual(result, 1000.0)
```
3 different test cases?

Stml89 replied:
Please divide this test into 3 TCs

VladislavSavchuk committed:
little fix and add logging to bank_deposit

VladislavSavchuk committed:
fix tests library

VladislavSavchuk committed:
add log that says that test bank was successful

VladislavSavchuk committed:
fix linters

VladislavSavchuk commented:
fixed according to the comments @Stml89

Stml89 commented on homework/homework20/test_bank_deposit.py:
```diff
@@ -0,0 +1,125 @@
+        assert bank.rate == 3.5, \
+            f"Expected rate to be 3.5, got {bank.rate}"
+
+        logger.info("Successfully verified bank attributes")
```
Nice

Stml89 replied:
add the same for other TCs as well

Stml89 commented on homework/homework20/test_bank_deposit.py:
```diff
@@ -0,0 +1,125 @@
+
+        bank = Bank(20000, 48, 5)
+        result = bank.calculate_monthly_capitalization()
+        self.assertEqual(result, 24417.91)
```
what is the diff between those 3 tests?

Stml89 commented on homework/homework20/test_bank_deposit.py:
```diff
@@ -0,0 +1,125 @@
+
+        bank = Bank(5000, 36, 4)
+        result = bank.calculate_monthly_capitalization()
+        self.assertNotEqual(result, 5000)
```
are they testing the case thing?

VladislavSavchuk committed:
fix tests bank deposit

VladislavSavchuk commented:
@Stml89
- corrected the tests based on your comments.
- divided the tests of boundary values into 3 separate test cases
- added Successfully logger to each test
- removed 2 unnecessary tests in test_calculate_monthly_capitalization that tested the same thing (my mistake)
- updated negative cases
It seems to be looking better now, thank you.
